### PR TITLE
feat(SOUNDS LIKE):  add "SOUNDS LIKE" syntax for String comparing.

### PR DESCRIPTION
**Pull request by neil4dong**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Add "SOUNDS LIKE" syntax for String comparing.
refer to this issue: https://github.com/datafuselabs/databend/issues/3042

Closes #issue

### CODE REVIEW

**Comment by sundy-li** [comment]
@neil4dong  

> I noticed SOUNDEX had already been implemented.
[expr1 SOUNDS LIKE expr2] is the same as [SOUNDEX(expr1) = SOUNDEX(expr2)]


If this is correct, I think you don't have to implement another function sounds like.
You can resolve the `Expr::SoundsLike` in type_checker like `Expr::Between`.


Between: Rewrite `expr BETWEEN low AND high`  into `expr >= low AND expr <= high`
SoundLike:  Rewrite `expr1 SOUNDS LIKE expr2`  into `SOUNDEX(expr1) =  SOUNDEX(expr2)`

**Comment by neil4dong** [comment]
> @neil4dong
> 
> > I noticed SOUNDEX had already been implemented.
> > [expr1 SOUNDS LIKE expr2] is the same as [SOUNDEX(expr1) = SOUNDEX(expr2)]
> 
> If this is correct, I think you don't have to implement another function sounds like. You can resolve the `Expr::SoundsLike` in type_checker like `Expr::Between`.
> 
> Between: Rewrite `expr BETWEEN low AND high` into `expr >= low AND expr <= high` SoundLike: Rewrite `expr1 SOUNDS LIKE expr2` into `SOUNDEX(expr1) = SOUNDEX(expr2)`

Yes, that is a better way to implement it.   I don't have a global view of the code ,thanks for the tips.

File: src/query/functions/tests/it/scalars/comparison.rs [hunk 1]
```diff
@@ -386,3 +387,21 @@ fn test_regexp(file: &mut impl Write) {
     run_ast(file, "lhs regexp rhs", &columns);
     run_ast(file, "lhs rlike rhs", &columns);
 }
+
+fn test_sounds_like(file: &mut impl Write) {
```
Comment (sundy-li) [inline src/query/functions/tests/it/scalars/comparison.rs:391]: could be removed.

You can add sqllogic tests to test `sounds like`.

**Comment by neil4dong** [comment]
why the main branch have lots of changes, did the team develop on the main branch ?   Some change looks not pass the checks.

File: src/query/functions/tests/it/scalars/comparison.rs (hunk 1 above)
Comment (neil4dong) [reply src/query/functions/tests/it/scalars/comparison.rs:391]: Is this test unnecessary? This it test is for AST sytax check . can we just keep it?

**Comment by sundy-li** [comment]
`env REGENERATE_GOLDENFILES=1 cargo test`

to regenerate the  test golden files.

**Comment by neil4dong** [comment]
> `env REGENERATE_GOLDENFILES=1 cargo test`
> 
> to regenerate the test golden files.

I removed the new added golden files test code. because I cannot build on my local machine（not enough memory）

**Review by sundy-li — APPROVED** [review]

**Comment by sundy-li** [comment]
Thanks for the contribution

**Pull request merged**